src/Nethermind/Nethermind.Core/BlockInfo.cs

```diff
@@ -14,27 +14,52 @@
 //  You should have received a copy of the GNU Lesser General Public License
 //  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
 
+using System;
 using Nethermind.Core.Crypto;
 using Nethermind.Int256;
 
 namespace Nethermind.Core
 {
+    [Flags]
+    public enum BlockMetadata
+    {
+        None = 0x0,
+        Finalized = 0x1,
+        Invalid = 0x2,
+    }
+
     public class BlockInfo
     {
         public BlockInfo(Keccak blockHash, in UInt256 totalDifficulty)
         {
             BlockHash = blockHash;
             TotalDifficulty = totalDifficulty;
         }
-        
+
         public UInt256 TotalDifficulty { get; }
-        
+
         public bool WasProcessed { get; set; }
-        
+
         public Keccak BlockHash { get; }
-        
-        public bool IsFinalized { get; set; }
-        
+
+        public bool IsFinalized
+        {
+            get => (Metadata & BlockMetadata.Finalized) == BlockMetadata.Finalized;
+            set
+            {
+                if (value)
+                {
+                    Metadata |= BlockMetadata.Finalized;
+                }
+                else
+                {
+                    Metadata &= ~BlockMetadata.Finalized;
+                }
+            }
+        }
```

Comment:
Sorry but I don't understand this code. You use a setter to change only another property
The only IsFinalized.set is on the Consensus.Aura project. Could it not be updated ?
Plus I think it's a dangerous practice, as `Metadata` can have technically its value modified in two different place (not on the property itself + difficulty to debug)

This code should be enough :
```suggestion
        public bool IsFinalized => (Metadata & BlockMetadata.Finalized) == BlockMetadata.Finalized;
```